Mathlib/MeasureTheory/Integral/FundThmCalculus.lean

```diff
@@ -1341,31 +1341,95 @@ section Parts
 
 variable [NormedRing A] [NormedAlgebra ℝ A] [CompleteSpace A]
 
-/-- For improper integrals, see `MeasureTheory.integral_deriv_mul_eq_sub`,
+/-- The integral of the derivative of a product of two maps.
+For improper integrals, see `MeasureTheory.integral_deriv_mul_eq_sub`,
 `MeasureTheory.integral_Ioi_deriv_mul_eq_sub`, and `MeasureTheory.integral_Iic_deriv_mul_eq_sub`. -/
-theorem integral_deriv_mul_eq_sub {u v u' v' : ℝ → A} (hu : ∀ x ∈ uIcc a b, HasDerivAt u (u' x) x)
-    (hv : ∀ x ∈ uIcc a b, HasDerivAt v (v' x) x) (hu' : IntervalIntegrable u' volume a b)
+theorem integral_deriv_mul_eq_sub_of_hasDeriv_right {u v u' v' : ℝ → A}
+    (hu : ContinuousOn u [[a, b]])
+    (hv : ContinuousOn v [[a, b]])
+    (huu' : ∀ x ∈ Ioo (min a b) (max a b), HasDerivWithinAt u (u' x) (Ioi x) x)
```

Comment:
What about also adding a version assuming `HasDerivAt u (u' x) x` (still for all `x \in Ioo`)? I believe this would be the most common application, so probably worth having a dedicated lemma.

Reply:
I think in many common applications you just have C^1 functions everywhere. Unless you want to prove a new lemma for an as general class as functions again (e.g. `fourierCoeffOn_of_hasDerivWithinAt`), and then you may want to generalize even further to this version. And your version is only one `.hasDerivWithinAt` away. Though granted, you sometimes have to write it as `(fun x hx => (h x hx).hasDerivWithinAt)` which is a bit cumbersome.
If you still think your version is worthwhile, I'll add it.